tpm2/tpm2_test.go

```diff
@@ -589,6 +590,59 @@ func TestPCREvent(t *testing.T) {
 	}
 }
 
+func TestPCRExtend(t *testing.T) {
+	rw := openTPM(t)
+	defer rw.Close()
+
+    const pcr = int(16)
+
+    // testing SHA1
```

Comment:
Refactor this into table-driven test with subtests. Something like:
```
tests := []struct {
  desc string
  hashAlg Algorithm
  hashSize int
  hashSum func([]byte) []byte
} {
  {
    desc: "SHA1",
    hashAlg: AlgSHA1,
    hashSize: sha1.Size,
    hashSum: func(in []byte) []byte {
      s := sha1.Sum(in)
      return s[:]
    }
  },
  {
    desc: "SHA256",
    ...
  }
}
for _, tt := range tests {
  t.Run(tt.desc, func(t *testing.T) {
    oldPCRValue, err := ...
  })
}
```

Reply:
yea this is cool - my limited go skills prevented me from thinking about something like that (y)